testssl.sh

```diff
@@ -17194,7 +17194,15 @@ EOF
 }
 
 maketempf() {
-     TEMPDIR=$(mktemp -d /tmp/testssl.XXXXXX) || exit $ERR_FCREATE
+     TEMPDIR=$(mktemp -d /tmp/testssl.XXXXXX)
+     if [[ $? -ne 0 ]]; then
+          # For e.g. devices where we can't write to /tmp we chose $PWD but we can't
+          # allow every char as we haven't quoted all strings depending on it, see #1445
+          if [[ $PWD =~ ^[A-Za-z0-9\.,-/_]+$ ]]; then
```

Comment:
Hi Dirk,

`^[A-Za-z0-9\.,-/_]+$` didn't work for me:
```
> [[ "/home/user" =~ ^[A-Za-z0-9\.,-/_]+$ ]] && echo "illegal charaters: \"$BASH_REMATCH\"" || echo "OK"
illegal charaters: "/home/user"
> [[ "/home/the user" =~ ^[A-Za-z0-9\.,-/_]+$ ]] && echo "illegal charaters: \"$BASH_REMATCH\"" || echo "OK"
OK
> [[ "/home/user/is a semicolon okay;" =~ ^[A-Za-z0-9\.,-/_]+$ ]] && echo "illegal charaters: \"$BASH_REMATCH\"" || echo "OK"
OK
```
`[[ "$PWD" =~ [^A-Za-z0-9\.,/_-] ]]` seems to work better:
```
> [[ "/home/user" =~ [^A-Za-z0-9\.,/_-] ]] && echo "illegal charaters: \"$BASH_REMATCH\"" || echo "OK"
OK
> [[ "/home/the user" =~ [^A-Za-z0-9\.,/_-] ]] && echo "illegal charaters: \"$BASH_REMATCH\"" || echo "OK"
illegal charaters: " "
> [[ "ls;" =~ [^A-Za-z0-9\.,/_-] ]] && echo "illegal charaters: \"$BASH_REMATCH\"" || echo "OK"
illegal charaters: ";"
> [[ "/home/user/this-is_my.path" =~ [^A-Za-z0-9\.,/_-] ]] && echo "illegal charaters: \"$BASH_REMATCH\"" || echo "OK"
OK
```